crates/backend/src/ast.rs

```diff
@@ -110,91 +123,144 @@ impl Hash for ImportModule {
     }
 }
 
+/// The type of item being imported
 #[cfg_attr(feature = "extra-traits", derive(Debug))]
 #[derive(Clone)]
 pub enum ImportKind {
+    /// Importing a function
     Function(ImportFunction),
+    /// Importing a static value
     Static(ImportStatic),
+    /// Importing a type/class
     Type(ImportType),
+    /// Importing a JS enum
     Enum(ImportEnum),
 }
 
+/// A function being imported from JS
 #[cfg_attr(feature = "extra-traits", derive(Debug))]
 #[derive(Clone)]
 pub struct ImportFunction {
+    /// The full signature of the function
     pub function: Function,
+    /// The name rust code will use
     pub rust_name: Ident,
+    /// The type being returned
     pub js_ret: Option<syn::Type>,
+    /// Whether to catch JS exceptions
     pub catch: bool,
+    /// Whether the function is variadic on the JS side
     pub variadic: bool,
+    /// Whether the function should use structural type checking
     pub structural: bool,
+    /// For testing purposes
```

Comment:
Could this be expanded with for what testing purposes this is used for?

(this is a public-facing attribute IIRC so it's not entirely just testing)

Reply:
Practically, it causes the Builder for the JS to bail if it finds itself generating code for something with `assert_no_shim`. I'll add an explanation of the behavior.